src/Operation/AbstractOperation.php

```diff
@@ -64,86 +63,53 @@ abstract class AbstractOperation
     private $connectTimeout = 5;
 
     /**
-     * Helper method to throw an exception when an API or HTTP endpoint responds
-     * with a non-200 status code.
-     *
-     * @param $request HttpRequest the HTTP request
-     * @param $response HttpResponse the HTTP response to check
-     * @return bool returns true when everything was okay
-     * @throws AbstractHttpException
-     */
-    protected static function checkResponse(HttpRequest $request, HttpResponse $response)
-    {
-        if ($response->getCode() !== 200) {
-            throw ExceptionBuilder::fromHttpRequestAndResponse($request, $response);
-        }
-        return true;
-    }
-
-    /**
-     * Create and returns a new API request object initialized with a content-type
-     * of 'application/json' and the specified authentication token
-     *
-     * @param Token|null $token the token to use for the endpoint
-     * @param string|null $nostoAccount
-     * @param string|null $domain
-     * @return ApiRequest the newly created request object.
-     * @throws NostoException if the account does not have the correct token set.
+     * @param Token|null $token
+     * @param null $nostoAccount
+     * @param null $domain
+     * @return ApiRequest|GraphqlRequest|HttpRequest
+     * @throws NostoException
      */
-    protected function initApiRequest(
+    protected function initRequest(
         Token $token = null,
         $nostoAccount = null,
         $domain = null
     ) {
         if (is_null($token)) {
             throw new NostoException('No API token found for account.');
         }
-        $request = new ApiRequest();
+        $request = $this->getRequestType();
         if (is_string($domain)) {
             $request->setActiveDomainHeader($domain);
```

Comment:
I would remove the mutability of `request`. Please ditch both setters and move to contructors.